### PR TITLE
Improve loading view and improve demo

### PR DESCRIPTION
**Pull request by 3lvis**
# Why?

Improved support for going from loading to success

# What?

Improves transition between states, this potentially could be refactored for better state handling but not sure if we should wait until we add "failed" for example.

# Show me

all

### CODE REVIEW

**Comment by 3lvis** [comment]
Okay, this time I'm pointing to the branch.

File: Sources/Fullscreen/LoadingView/Demo/LoadingViewDemoView.swift [hunk 1]
```diff
@@ -22,18 +69,37 @@ public class LoadingViewDemoView: UIView {
...
 
     private func setup() {
-        addSubview(startLoading)
-        NSLayoutConstraint.activate([
-            startLoading.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -.largeSpacing),
-            startLoading.centerXAnchor.constraint(equalTo: centerXAnchor)
-            ])
+        addSubview(tableView)
+        tableView.fillInSuperview()
+        tableView.dataSource = self
```
Comment (robinsalehjan) [inline Sources/Fullscreen/LoadingView/Demo/LoadingViewDemoView.swift]: Nitpicking, but can we move this into the computed `tableView` property?

File: Sources/Fullscreen/LoadingView/Demo/LoadingViewDemoView.swift [hunk 2]
```diff
@@ -22,18 +69,37 @@ public class LoadingViewDemoView: UIView {
...
+        tableView.delegate = self
```
Comment (robinsalehjan) [inline Sources/Fullscreen/LoadingView/Demo/LoadingViewDemoView.swift]: And this one as well.

**Comment by 3lvis** [comment]
Adressed PR comments :)

**Comment by 3lvis** [comment]
🎉

**Pull request merged**